README.md

```diff
@@ -1,2 +1,95 @@
+
+## Setup
+
+Install the project dependencies with `poetry`, using Python 3.10 (you can use [pyenv](https://github.com/pyenv/pyenv) to manage multiple Python versions):
+
+```bash
+python3.10 -m pip install poetry
+python3.10 -m poetry install
+python3.10 -m poetry shell
+```
+
+Copy `.env.example` to `.env` and fill in the values:
+
+### OpenAI API key
+
+We will provide you with OpenAI key that's allowed to use gpt-3.5-turbo and embedding models, contact us on the TG group above.
+
+However, everyone is welcome to use arbitrary LLM if wanted.
+
+### Tavily API key
+
+Create a free acount on https://tavily.com and get the key there.
+
+Again, everyone is welcome to use arbitrary search engines, combine them, or even do a totally different approaches!
+
+### Private key on Gnosis Chain
+
+Use your existing or create a new wallet on Gnosis Chain. 
+
+By default the script will do only very tiny bets (0.00001 xDai per market), but of course, you can contact us on the TG group above with your public key to get some free xDai.
+
+## Task
+
+Your task is to modify `predict` function in `trader/prediction.py` by any means necessary.
+
+Goal of the `predict` function is, given an `question` about the future, answer it with either `True` (the answer is `yes`), `False` (if the answer is `no`) or `None` (if the prediction failed).
+
+All the questions are guaranteed to be about the future and to be in a binary yes/no format.
+
+You can play with the prompts, different approaches, different LLMs, search engines, or anything you can think of.
+
+The code can be messy, the only thing we ask you is for it to be reproducible on our machines, and to help with that, there is `mypy` as the only check of CI pipeline on Github.
+
+### Experimenting
+
+Run 
+
+```bash
+PYTHONPATH=. streamlit run trader/app.py
+```
+
+to start a Streamlit application where you can give your prediction method either question [from the Omen market](https://aiomen.eth.limo/), or write your own.
+
+Run 
+
+```bash
+python trader/benchmark.py --n N
+```
+
+where `N` is number of markets to do a prediction on. The benchmark script will run
+
+1. Random agent (coin flip between yes and no answers)
+2. Question-only agent (only LLM call, without any information from internet)
+3. `prediction.py/predict`-based agent
+
+on `N` open markets from https://manifold.markets. 
+
+The idea is that markets on Manifold are mostly answered by real people, so the closer your agent is to their predictions, the better. However, it isn't always the case.
+
+Bear in mind your LLM credits, Tavily credits or any other paid 3rd provider credits when running the benchmark, as it answers many markets in a single run, which can be very costly.
+
+### Submission
+
+1. Run `python trader/main.py`, it will place bets on all markets that will be used for the evaluation. You can run the script multiple times, but we will always look only at the latest bet on the market from your public key.
```

Comment:
I wanted us to manually curate a set of markets to open before the end of June and then ping them on TG that it's safe to run the agents.

> could just answer them manually themselves

I guess they could always do this, can't they? That'd be the `cheating` part of the evaluation.

> one created before the date of the hackathon

Maybe the `main.py` script could have a `--final` flag? By default final=False and it will bet on any open market on Omen. But with final=True, it will filter only for the curated list of markets used for the winner evaluation.

Reply:
> I guess they could always do this, can't they?

Not if the markets aren't created til after the submission

> Maybe the main.py script could have a --final flag?

That seems like a reasonable middleground to discourage people from gaming the competition by re-running until their agent gets 'lucky' (i.e. no crappy predictions, based on just eye-balling) 👍